#143 Additional phone numbers using ArrayField

vinyasmusic opened the pull request:
### Issue Reference
This PR addresses the Issue : Fixes #143 

### Summarize
Added a field in model called 'additional_phone_numbers' which is an ArrayField (taken from Postgres fields).
After save the values are being checked and cleaned  for valid phone numbers by overriding save logic


Edit : Please check the Malayalam translation for the field. I used Google translate

yeaske commented:
Shouldn't there be UI html changes associated with this to input these values?  Or is that already handled?

vinyasmusic commented:
Added the migrations. 
I have added the field in the fields column in CreateRequest view. Works just like the other fields, the logic to separate out the phone numbers is handled in the save func in model.

vinyasmusic commented:
@yeaske For showing the additional phone numbers in Request details page should I do something like this 
`Additional phone numbers :
9898989898, 9898989898, 9898989898`
or 
`Additional phone numbers :
9898989898
9898989898
9898989898`

vinyasmusic commented:
Added for the detail view

ragsagar commented on mainapp/models.py:
```diff
@@ -109,11 +114,20 @@ def summarise(self):
+    def save(self, *args, **kwargs):
+        if self.additional_phone_numbers:
+            self.additional_phone_numbers = [item.strip() for item in self.additional_phone_numbers[0].split(' ')
+                                             if item is not '' and re.match("^\+?1?\d{9,15}$", item) is not None]
```
You can use `if not item` instead of `if item is not ''`

vinyasmusic replied:
I wanted it to be explicit check for '' since if there are multiple spaces in between two numbers then we get '' after split.

salmanulfarzy commented on mainapp/models.py:
```diff
@@ -91,6 +94,8 @@ class Request(models.Model):
         default = 'new'
     )
     supply_details = models.CharField(max_length=100, blank=True)
+    additional_phone_numbers = ArrayField(models.CharField(max_length=200), blank=True, default=[],
```
Isn't it [recommended](https://docs.djangoproject.com/en/2.1/ref/contrib/postgres/fields/#django.contrib.postgres.fields.ArrayField) to use `list` instead of `[]` as `default` for `ArrayField` ?

vinyasmusic replied:
Updated, please review

salmanulfarzy commented on mainapp/models.py:
```diff
@@ -117,6 +120,8 @@ class Request(models.Model):
         default = 'new'
     )
     supply_details = models.CharField(max_length=100, blank=True)
+    additional_phone_numbers = ArrayField(models.CharField(max_length=200), blank=True, default=list,
```
Why `max_length` to 200, We could do with a much smaller number for two phone numbers right ?

vinyasmusic replied:
I did not want to restrict to two phone numbers hence the choice of 200. Can be reduced to 100 which would mean a person can put in 9 phone numbers at the max.
